wrangler-core/src/main/java/io/cdap/directives/parser/FixedLengthParser.java

```diff
@@ -148,4 +153,32 @@ public Mutation lineage() {
       .all(Many.of(col), Many.of(col))
       .build();
   }
+
+  @Override
+  public Relation transform(RelationalTranformContext relationalTranformContext,
+                            Relation relation) {
+    java.util.Optional<ExpressionFactory<String>> expressionFactory = SqlExpressionGenerator
+            .getExpressionFactory(relationalTranformContext);
+    if (!expressionFactory.isPresent()) {
+      return new InvalidRelation("Cannot find an Expression Factory");
+    }
+
+    int currentpos = 1;
+    int columncounter = 1;
+
+    for (int width : widths) {
+      relation = relation.setColumn(String.format("%s_%d", col, columncounter),
+              expressionFactory.get().compile(String
+                      .format("replace(substr(%s, %d, %d), '%s', \"\")"
```

Comment:
What happens when widths are greater than the length of the string?

Reply:
In that case the current directive behavior filters out that record. If the width is greater than the length of all the values in that column, no data is returned. To replicate the same behavior I have added the additional filter condition.